core/kazoo_globals/src/kz_globals.erl

```diff
@@ -412,70 +425,149 @@ maybe_add_zone(Zone, #state{zones=Zones}) ->
+        T1 when N1 =:= N2 -> 'no';
+        T1 when T1 < T2 -> 'no';
+        T1 when T1 > T2 -> 'yes';
+        T1 when T1 =:= T2 andalso N1 < N2 -> 'no';
+        T1 when T1 =:= T2 andalso N1 > N2 -> 'yes'
+    end.
+
+-spec register_local(kz_global:global()) -> 'yes' | 'no'.
+register_local(Global) ->
+    case where(kz_global:name(Global)) of
+        Global ->
+            Updated = kz_global:register_local(?TAB_NAME, Global),
+            advertise_register(Updated),
+            'yes';
+        _ ->
+            lager:debug("cannot register ~p locally", [kz_global:name(Global)]),
+            'no'
+    end.
 
 -spec advertise_register(kz_global:global()) -> 'ok'.
--spec advertise_register(kz_global:name(), ne_binary()) -> 'ok'.
 advertise_register(Global) ->
-    advertise_register(kz_global:name(Global)
-                       ,kz_global:server(Global)
-                      ).
-
-advertise_register(Name, Queue) ->
-    Payload = [{<<"Name">>, Name}
-              ,{<<"State">>, 'local'}
-               | kz_api:default_headers(Queue, ?APP_NAME, ?APP_VERSION)
+    Payload = [{<<"Name">>, kz_global:name(Global)}
+              ,{<<"State">>, 'registered'}
+              ,{<<"Timestamp">>, kz_global:timestamp(Global)}
+               | kz_api:default_headers(kz_global:server(Global), ?APP_NAME, ?APP_VERSION)
               ],
     kz_amqp_worker:cast(Payload, ?AMQP_REGISTER_FUN).
 
--spec register_remote(kz_global:global()) -> pid().
-register_remote(Global) ->
-    {Pid, Ref} = start_proxy(Global),
-    ProxyGlobal = kz_global:update_with_pid_ref(Global, Pid, Ref),
-    lager:debug("inserting proxy ~p", [ProxyGlobal]),
-    ets:insert(?TAB_NAME, ProxyGlobal),
-    Pid.
+-spec register_remote(kz_global:global(), term()) -> 'ok'.
+register_remote(Global, From) ->
+    register_remote(where(kz_global:name(Global)), Global, From).
+
+-spec register_remote(kz_global:global() | 'undefined', kz_global:global(), term()) -> 'ok'.
+register_remote('undefined', Global, From) ->
+    ProxyGlobal = kz_global:register_remote(?TAB_NAME, Global),
+    maybe_register_remote_reply(From, kz_global:pid(ProxyGlobal));
+register_remote(Local, Remote, From) ->
+    case register_remote_resolve(Local, Remote) of
+        'no' ->  maybe_register_remote_reply(From, kz_global:pid(Local));
+        'yes' ->
+            ProxyGlobal = kz_global:register_remote(?TAB_NAME, Remote),
+            maybe_register_remote_reply(From, kz_global:pid(ProxyGlobal))
+    end.
+
+-spec register_remote_resolve(kz_global:global(), kz_global:global()) -> 'yes' | 'no'.
+register_remote_resolve(Local, Remote) ->
+    T1 = kz_global:timestamp(Local),
+    T2 = kz_global:timestamp(Remote),
+    N1 = kz_global:node(Local),
+    N2 = kz_global:node(Remote),
+    case T1 of
+        T1 when N1 =:= N2 -> 'no';
+        T1 when T1 < T2 -> 'no';
+        T1 when T1 > T2 -> 'yes';
+        T1 when T1 =:= T2 andalso N1 < N2 -> 'no';
+        T1 when T1 =:= T2 andalso N1 > N2 -> 'yes'
+    end.
```

Comment:
@lazedo i was planning on addressing this in [KAZOO-4783](https://2600hz.atlassian.net/browse/KAZOO-4783) since it was out of scope of this PR. Can we revert the commit here and address it in the new ticket?